### PR TITLE
INTYGFV-14291: Renew body now supports both FK/AG7804. Small text cha…

### PR DESCRIPTION
**Pull request by RobertDanielsson**
…nge in FK7804 body.

The text toggle in LISJP text is according to this: https://inera.atlassian.net/wiki/spaces/IT/pages/9819062/D1.4.1.+Knappar+i+intygsheader#D1.4.1.Knappariintygsheader-modal1.6

### CODE REVIEW

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetCertificatesAvailableFunctionsImpl.java [hunk 1]
```diff
@@ -412,6 +401,49 @@ public GetCertificatesAvailableFunctionsImpl(AuthoritiesHelper authoritiesHelper
         return resourceLinks;
     }
 
+    private String getRenewBody(Certificate certificate) {
+        if (isLisjpCertificate(certificate)) {
+            final var complementaryText =
+                isUserAndCertificateFromSameCareUnit(certificate) ? "Eventuell kompletteringsbegäran kommer att klarmarkeras."
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetCertificatesAvailableFunctionsImpl.java:407]: Would be great if the logic deciding the message, for renewal, is covered by unit tests.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetCertificatesAvailableFunctionsImpl.java (hunk 1 above)
Comment (mikeyOne) [reply web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetCertificatesAvailableFunctionsImpl.java:407]: The check for same or different care unit seems to be based on **unit or care unit** (whichever is present in valdVardEnhet and unit). Apparently there are some confusion in requirements regarding unit vs care unit, but requirements says the check should be on care unit. So, perhaps have another look at whether the check should be strictly on care unit or not...

If strictly care unit there might be useable code in PR 14151

Comment (mww-sopra-steria) [reply web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetCertificatesAvailableFunctionsImpl.java:407]: Good catch @mikeyOne !

**Review by mww-sopra-steria — APPROVED** [review]
Nice!

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -104,7 +104,7 @@ private String getRole(WebCertUser webCertUser) {
             : webCertUser.getRoleTypeName().split(" ")[0];
     }
 
-    private Vardenhet getLoggedInCareUnit(WebCertUser webCertUser) {
+    public Vardenhet getLoggedInCareUnit(WebCertUser webCertUser) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/UserServiceImpl.java:107]: Add Override annotation

**Review by mww-sopra-steria — APPROVED** [review]

**Pull request merged**